neuralmonkey/decoding_function.py

```diff
@@ -47,6 +48,12 @@ def attention_decoder(decoder_inputs, initial_state, attention_objects,
             outputs.append(output)
             states.append(state)
 
+        if summary_collections:
+            for i, a in enumerate(attention_objects):
+                alignments = tf.expand_dims(tf.transpose(tf.pack(a.attentions_in_time), perm=[1, 2, 0]), -1)
```

Comment:
I wonder if this is general enough. If you look e.g, at the [Recurrent Neural Machine Translation](https://arxiv.org/abs/1607.08725) paper, they use GRU net do the attention instead of the softmax weighting and come with a clever way visualizing this attention. i think that reserving the `attention_in_time` field for the visualization purpose should work even for the recurrent attention, but please check if it is so.